pkg/api/customization/workload/workload.go

```diff
@@ -124,6 +127,16 @@ func update(apiContext *types.APIContext, data map[string]interface{}, ID string
 
 func (a ActionWrapper) rollbackDeployment(apiContext *types.APIContext, clusterContext *config.UserContext,
 	actionName string, deployment projectclient.Workload, namespace string, name string) error {
+	split := strings.SplitN(apiContext.ID, ":", 3)
+	// apiContext.ID is of the form deployment:ns:name
+	if len(split) != 3 || split[0] != appsv1.DeploymentResource.SingularName {
+		return httperror.NewAPIError(httperror.InvalidBodyContent, fmt.Sprintf("invalid deployment %s", apiContext.ID))
+	}
+	deploymentNamespace, deploymentName := split[1], split[2]
+	depl, err := clusterContext.Apps.Deployments("").Controller().Lister().Get(deploymentNamespace, deploymentName)
```

Comment:
I think we should do this after validation of input, ie ParseAndValidateActionBody as well as validation on revision